### ISSUE TITLE
Training seems to be very slow on Tesla K40

### ISSUE THREAD

**Issue by michaelhuang74** [lengstrom/fast-style-transfer#22]
I am trying to generate a model file using the following,

python style.py --style  examples/style/la_muse.jpg --checkpoint-dir checkpoint/check2  --content-weight 1.5e1 --checkpoint-iterations 1 --batch-size 20

**At very beginning of the training process, it took about 20 seconds to finish one iteration. However, as the training process goes on, the time of one iteration becomes much longer. Currently the training process has been going on for more than 14 hours and only 621 iterations have finished.** 

The followings are the screen output during the training process.

I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcurand.so locally
Train set has been trimmed slightly..
(1, 512, 512, 3)

I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:925] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
I tensorflow/core/common_runtime/gpu/gpu_device.cc:951] Found device 0 with properties: 
name: Tesla K40c
major: 3 minor: 5 memoryClockRate (GHz) 0.745
pciBusID 0000:01:00.0
Total memory: 11.17GiB
Free memory: 11.10GiB
W tensorflow/stream_executor/cuda/cuda_driver.cc:572] creating context when one is currently active; existing: 0x2a55840
I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:925] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
I tensorflow/core/common_runtime/gpu/gpu_device.cc:951] Found device 1 with properties: 
name: Tesla K40c
major: 3 minor: 5 memoryClockRate (GHz) 0.745
pciBusID 0000:02:00.0
Total memory: 11.17GiB
Free memory: 11.10GiB
I tensorflow/core/common_runtime/gpu/gpu_device.cc:972] DMA: 0 1 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 0:   Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 1:   Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Tesla K40c, pci bus id: 0000:01:00.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:1) -> (device: 1, name: Tesla K40c, pci bus id: 0000:02:00.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Tesla K40c, pci bus id: 0000:01:00.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:1) -> (device: 1, name: Tesla K40c, pci bus id: 0000:02:00.0)
UID: 11
Epoch 0, Iteration: 1, Loss: 1.60785e+08
style: 1.4523e+08, content:8.83568e+06, tv: 6.71974e+06
Epoch 0, Iteration: 2, Loss: 1.08762e+08
style: 9.43464e+07, content:7.88181e+06, tv: 6.5336e+06
.............................................
Epoch 0, Iteration: 620, Loss: 1.08479e+07
style: 3.26121e+06, content:6.84982e+06, tv: 736917.0
Epoch 0, Iteration: 621, Loss: 1.12474e+07
style: 3.23448e+06, content:7.27868e+06, tv: 734276.0

**Only 621 iteratons have finished after more than 14 hours. Currently, one iteration takes a couple of minutes. 

Any idea why the training process is so slow on Tealsa K40?
In addition, how can I adjust the parameters (e.g., batch-size) in order to speedup the training process without compromising the quality of the model file too much? Thanks.**

**Comment by lengstrom** [comment]
`checkpoint_iterations=1` forces your program to save a checkpoint every iteration, so you should make that parameter larger if you want to train a reasonable amount of time.

**Comment by michaelhuang74** [comment]
@lengstrom Thanks. I will change checkpoint_iterations to a larger number.

Additionally, I have a question regarding when the training is supposed to finish.
(1) Does the training process go through a fixed number of iterations, say, 10,000? If the training goes through a fixed number of iterations, where is the number of iterations defined? I went through style.py, but could not find this information. I am new to tensorflow and might overlook the code.
(2) If the above is not true, under what kind of condition does the training process finish? 

Thanks.

**Comment by lengstrom** [comment]
Look at the epochs flag under `docs.md`.

**Comment by jinaviva** [comment]
Amazing project! Must use the Unix like OS（ Linux operating system） to running tensorflow?Can i use VMware Linux?Is there any suggestion. sorry,im a noob!

**Comment by lengstrom** [comment]
I don't know, I've never used VMWare, sorry!
ᐧ

**Comment by Zulqurnain24** [comment]
Hello,
I have mac with following specifications:

MacBook Air (13-inch, Early 2015)
1.6 GHz Intel Core i5
4 GB 1600 MHz DDR3
Intel HD Graphics 6000 1536 MB

I have executed it but the process is really slow even for lesser than 1000 iterations, does Intel HD Graphics 6000 1536 MB supports cuda tensorflow or i need to buy a MacBook pro with radeon 450 or above?
